src/m365/spo/commands/folder/folder-roleassignment-add.ts

```diff
@@ -145,20 +162,32 @@ class SpoFolderRoleAssignmentAddCommand extends SpoCommand {
       }
 
       const roleDefinitionId = await this.getRoleDefinitionId(args.options);
+      let principalId: number | undefined = args.options.principalId;
       if (args.options.upn) {
-        const upnPrincipalId = await this.getUserPrincipalId(args.options);
-        await this.breakRoleAssignment(requestUrl);
-        await this.addRoleAssignment(requestUrl, upnPrincipalId, roleDefinitionId);
+        principalId = await this.getUserPrincipalId(args.options);
       }
       else if (args.options.groupName) {
-        const groupPrincipalId = await this.getGroupPrincipalId(args.options);
-        await this.breakRoleAssignment(requestUrl);
-        await this.addRoleAssignment(requestUrl, groupPrincipalId, roleDefinitionId);
+        principalId = await this.getGroupPrincipalId(args.options);
       }
-      else {
-        await this.breakRoleAssignment(requestUrl);
-        await this.addRoleAssignment(requestUrl, args.options.principalId!, roleDefinitionId);
+      else if (args.options.entraGroupId || args.options.entraGroupName) {
+        if (this.verbose) {
+          await logger.logToStderr('Retrieving group information...');
+        }
+
+        let group: Group;
+        if (args.options.entraGroupId) {
+          group = await entraGroup.getGroupById(args.options.entraGroupId);
+        }
+        else {
+          group = await entraGroup.getGroupByDisplayName(args.options.entraGroupName!);
+        }
+
+        const siteUser = await spo.ensureEntraGroup(args.options.webUrl, group);
+        principalId = siteUser.Id;
       }
+
+      await this.breakRoleAssignment(requestUrl);
```

Comment:
woooot this command not only adds role assignment but breaks inheritance if it was not broken 😮
TBH I was not aware of it and the command description does not mention it I think.
@pnp/cli-for-microsoft-365-maintainers I think it was our mistake and it is a bug as we have separate command for breaking role inheritance and no other role assignment command for either file or list o list item will break the role inheritance.